src/dimension.c

```diff
@@ -1619,13 +1631,12 @@ dimension_rename_schema_name(TupleInfo *ti, void *data)
 
 /* Go through internal dimensions table and rename all relevant schema */
 void
-ts_dimensions_rename_schema_name(char *old_name, char *new_name)
+ts_dimensions_rename_schema_name(const char *old_name, const char *new_name)
 {
 	NameData old_schema_name;
 	ScanKeyData scankey[1];
 	Catalog *catalog = ts_catalog_get();
-	char *names[2] = { old_name, new_name };
-
+	char *names[2] = { pstrdup(old_name), pstrdup(new_name) };
```

Comment:
why do we need to copy these now?

Reply:
because `old_name` and `new_name` are now `const char *` pointers but scanning function expects a non-const data argument. I could cast away the `const`, but it seems to me that this is safer and prevents bugs in case of future changes.

Reply:
Since `ScannerCtx scanctx` is in the stack then `.data = names,` will survive only until the function call returns. In that case shouldn't we explicitly deallocate at the end of the function instead of relying on the current memory context?

Reply:
@mfundul it is standard practice in PostgreSQL to defer deallocation of palloc'd objects to the point when the memory context is deleted. This happens all the time. The only thing to be concerned about is if we would allocate unbounded memory, e.g., in a loop over an undefined number of objects.  This is not the case here however.

Reply:
This was turned into a cast per request.